k8s/client.go

```diff
@@ -16,27 +25,79 @@ type Client interface {
 	WatchIngresses(Watcher) error
 }
 
-type noopClient struct {
+type impl struct {
+	baseURL string
+	caCert  []byte
+	token   string
+	client  *http.Client
 }
 
-// NewClient creates a new K8 Client
-func NewClient() Client {
-	return &noopClient{}
-}
+// New creates a client for the kubernetes apiserver.
+func New(apiServerURL string, caCert []byte, token string) (Client, error) {
+	parsedURL, err := url.Parse(apiServerURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid url %s: %v", apiServerURL, err)
+	}
+	baseURL := strings.TrimSuffix(parsedURL.String(), "/")
+
+	pool := x509.NewCertPool()
+	pool.AppendCertsFromPEM(caCert)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{RootCAs: pool},
+	}
+	client := &http.Client{Transport: tr}
+
+	log.Debugf("Constructing client with url: %s, token: %s, caCert: %v",
+		baseURL, token, string(caCert))
 
-func (client *noopClient) GetIngresses() ([]Ingress, error) {
-	return []Ingress{}, nil
+	return &impl{
+			baseURL: baseURL,
+			caCert:  caCert,
+			token:   token,
+			client:  client},
+		nil
 }
 
-func (client *noopClient) WatchIngresses(w Watcher) error {
-	log.Infof("Watching ingresses {}", w)
+func (i *impl) GetIngresses() ([]Ingress, error) {
+	endpoint := i.baseURL + "/apis/extensions/v1beta1/ingresses"
+	req, err := http.NewRequest("GET", endpoint, nil)
```

Comment:
Yup I already fixed this in my other branch. I can fix it here though - it was the main cause of my problems trying to debug why it wasn't working :).